web-react-ts/src/pages/campaigns/multiplication/MulltiplicationCampaignServiceForm.tsx

```diff
@@ -0,0 +1,353 @@
+    cediIncome: Yup.number()
+      .typeError('Please enter a valid number')
+      .positive()
+      .required('You cannot submit this form without entering your income'),
+    foreignCurrency: Yup.string(),
+    souls: Yup.number()
+      .typeError('Please enter a valid number')
+      .integer('You cannot enter decimals here')
+      .required(
+        'You cannot submit this form without entering your number of souls won'
+      ),
+    miracles: Yup.number()
+      .typeError('Please enter a valid number')
+      .integer('You cannot enter decimals here')
+      .required(
+        'You cannot submit this form without entering your number of miracles'
+      ),
+    crusadePictures: Yup.array()
+      .min(10, 'You must have at least 10 event pictures')
+      .of(Yup.string().required('You must have at least 10 event pictures')),
+    treasurerSelfie: Yup.string().required('You must take a treasurers selfie'),
+    treasurers: Yup.array()
+      .min(2, 'You must have at least two treasurers')
+      .of(Yup.string().required('Please pick a name from the dropdown')),
+  })
+
+  const checkIfArrayHasRepeatingValues = (array: any[]) => {
+    const sortedArray = array.sort()
+    for (let i = 0; i < sortedArray.length - 1; i++) {
+      if (sortedArray[i + 1] === sortedArray[i]) {
+        return true
+      }
+    }
+    return false
+  }
+
+  const parseForeignCurrency = (string: string) => {
+    const nonOptions = [
+      0,
+      '',
+      'o',
+      '0',
+      '00',
+      '0.0',
+      '0.00',
+      'Gh',
+      'ghana cedis',
+      'No',
+      'None',
+      'n/a',
+      'N/A',
+      'NA',
+      '-',
+    ]
+
+    if (nonOptions.includes(string.toLowerCase().trim())) {
+      return null
+    }
+
+    return null
+  }
+
+  const onSubmit = (
+    values: FormOptions,
+    onSubmitProps: FormikHelpers<FormOptions>
+  ) => {
+    if (checkIfArrayHasRepeatingValues(values.treasurers)) {
+      throwToSentry('You cannot choose the same treasurer twice!')
+      onSubmitProps.setSubmitting(false)
+      return
+    } else {
+      onSubmitProps.setSubmitting(true)
+      RecordServiceMutation({
+        variables: {
+          churchId: churchId,
+          crusadeDate: values.crusadeDate,
+          preacherId: values?.preacher,
+          crusadeLocation: values?.crusadeLocation,
+          attendance: parseInt(values.attendance),
+          income: parseFloat(values.cediIncome),
+          foreignCurrency: parseForeignCurrency(values.foreignCurrency),
+          souls: parseInt(values.souls),
+          treasurers: values?.treasurers,
+          treasurerSelfie: values.treasurerSelfie,
+          miracles: parseInt(values.miracles),
+          crusadePictures: values.crusadePictures,
+        },
+      })
+        .then((res) => {
+          onSubmitProps.setSubmitting(false)
+          onSubmitProps.resetForm()
+          clickCard(res.data.RecordService)
+          alert('success')
```

Comment:
Also, this success message is not the best. You should write something more descriptive